Frontend/src/Pages/ScriptEdition/ScriptEdition.jsx

```diff
@@ -105,7 +121,11 @@ export default function ScriptEdition(props) {
         socket.on('delete replica', async (replica) => {
             removeReplica(replica._id);
         });
-    }, [replicas, setReplicas, updateReplica, removeReplica, socket])
+        socket.on('update generation status', async({generationStatus}) => {
+            updateGenerationButtons({generationStatus});
```

Comment:
```suggestion
            updateGenerationButtons(generationStatus);
```
Explique moi l'utilité des {} stp